packages/react-native/src/theme/types.ts

```diff
@@ -0,0 +1,61 @@
+import { PartialDeep } from 'type-fest';
+import { ColorValue } from '@aws-amplify/ui';
+
+interface BaseTokens {
+  colors: Partial<Colors>;
+}
+
+export interface Tokens extends BaseTokens {
+  components: ComponentStyles;
+}
+
+type Override = Omit<StrictTheme, 'overrides'>;
+
+export type ColorMode = 'light' | 'dark' | 'system';
+
+/**
+ * A Theme just needs a name.
+ * Users can define any tokens they need, but they don't need a complete theme with all tokens.
+ */
+export interface Theme {
+  colorMode?: ColorMode;
+  /**
+   * The name of the theme.
+   */
+  name: string;
+  /**
+   * Component and component agnostic tokens.
+   */
+  tokens?: PartialDeep<Tokens>;
```

Comment:
Can we just declare `Tokens` in way that is loose and then wrap it with `Required` for the strict theme?

Reply:
Required is not deep